packages/grafana-ui/src/components/ElementSelectionContext/ElementSelectionContext.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { createContext, useCallback, useContext } from 'react';
+
+/** @alpha */
+export interface ElementSelectionContextState {
+  /**
+   * Turn on selection mode & show selection state
+   */
+  enabled?: boolean;
+  /** List of currently selected elements */
+  selected: ElementSelectionContextItem[];
+  onSelect: (item: ElementSelectionContextItem, multi?: boolean) => void;
+}
+
+export interface ElementSelectionContextItem {
+  id: string;
+}
+
+export const ElementSelectionContext = createContext<ElementSelectionContextState | undefined>(undefined);
+
+export interface UseElementSelectionResult {
+  isSelected?: boolean;
+  isSelectable?: boolean;
+  onSelect?: (evt: React.PointerEvent) => void;
+}
+
+export function useElementSelection(id: string | undefined): UseElementSelectionResult {
+  if (!id) {
+    return {};
+  }
+
+  const context = useContext(ElementSelectionContext);
+  if (!context) {
+    return {};
+  }
+
+  const isSelected = context.selected.some((item) => item.id === id);
+  const onSelect = useCallback<React.PointerEventHandler>(
```

Comment:
I think this is breaking the rules of hooks again.

Reply:
not really, if is no context and a context is added there will be an unmount/remount 

so an early exit when there is no context is fine